src/lib/y2storage/partition_tables/base.rb

```diff
@@ -114,6 +115,26 @@ def inspect
         "<PartitionTable #{self}[#{num_children}] #{parts}#{slots}>"
       end
 
+      # The partition id depends on the partition table type. For example,
+      # MSDOS partition tables use SWAP id for swap partitions, but DASD
+      # partiton tables always need LINUX id.
+      #
+      # With this method, each partition table can define the partition id
+      # it expects for each case.
+      #
+      # TODO: right now only :swap is considered.
+      #
+      # @param id_name [Symbol] a partition id name (e.g., :swap, :linux, :ntfs)
+      # @return [PartitionId]
+      def partition_id_for(id_name)
+        case id_name
+        when :swap
+          PartitionId::SWAP
+        else
```

Comment:
Hmmm, if only `:swap` is considered... why do we have an `else` here? Not sure if I like how the whole situation of inheritance and this method is resolved.